packages/components/src/styles/components/stepper/list.scss

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+//
+// STEPPER > LIST
+//
+
+$hds-stepper-list-step-padding-bottom: 16px;
+
+.hds-stepper-list,
+.hds-stepper-list__step {
+  margin: 0;
+  padding: 0;
+  list-style: none;
+}
+
+// SUB COMPONENT: STEPPER > LIST > STEP
+
+.hds-stepper-list__step {
+  display: grid;
+  grid-column-gap: 8px;
+  grid-template-columns: auto 1fr;
+
+  &:not(:last-of-type) {
+    padding-bottom: $hds-stepper-list-step-padding-bottom;
+
+    .hds-stepper-list__step__progress {
+      &::before,
+      &::after {
+        position: absolute;
+        top: 1px;
+        left: 11px;
+        width: 2px;
+        content: "";
+      }
+
+      &::before {
+        height: calc(100% + $hds-stepper-list-step-padding-bottom);
+        background-color: var(--token-color-palette-alpha-200);
+      }
+
+      &::after {
+        z-index: -1;
+        background-color: var(--token-color-palette-neutral-700);
+      }
+    }
+  }
+}
+
+.hds-stepper-list__step__progress {
```

Comment:
I commented on an individual commit but repeating here for easier reference.

In BEM naming conventions, it's recommended only going one level deep in the naming structure for child components, so class names should only contain one "__" max.